.github/workflows/build.yml

```diff
@@ -31,25 +31,46 @@ jobs:
         python-version: ["3.10"]
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v3
+      - name: Connect to Container Registry
+        uses: docker/login-action@v2.1.0
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Python info
-        shell: bash -e {0}
-        run: |
-          which python3
-          python3 --version
-      - name: Upgrade pip and install dependencies
-        run: |
-          python3 -m pip install --upgrade pip poetry
-          poetry install --with test
-      - name: Run unit tests
-        run: poetry run pytest -v tests/unit_tests
-      - name: Run e2e tests
-        run: poetry run pytest -v --headless tests/e2e_tests
-      - name: Verify that we can build the package
-        run: poetry build
+          registry: ghcr.io
+          username: wbaccinelli
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run Tests in Container
+        run: >
+          docker run
+          --rm
+          -v ${{ github.workspace }}:/ci
+          -e TEST_DATA=/eitprocessing
+          ghcr.io/eit-alive/eittestdata:latest
+          sh -c 'set -xe ;
+            cd /ci ;
+            python3 -m pip install --upgrade pip poetry ;
+            poetry install --with test ;
+            poetry run pytest -v tests/unit_tests ;
+            python -m build'
+#    steps:
+#      - uses: actions/checkout@v3
+#      - name: Set up Python ${{ matrix.python-version }}
+#        uses: actions/setup-python@v3
+#        with:
+#          python-version: ${{ matrix.python-version }}
+#      - name: Python info
+#        shell: bash -e {0}
+#        run: |
+#          which python3
+#          python3 --version
+#      - name: Upgrade pip and install dependencies
+#        run: |
+#          python3 -m pip install --upgrade pip poetry
+#          poetry install --with test
+#      - name: Run unit tests
+#        run: poetry run pytest -v tests/unit_tests
+#      - name: Run e2e tests
+#        run: poetry run pytest -v --headless tests/e2e_tests
+#      - name: Verify that we can build the package
+#        run: poetry build
```

Comment:
why is this commented out rather than deleted?

Reply:
I forgot to remove this (I had no hope that the test would have worked!). I removed this, and I had to remove the e2e test form the workflow, because I tried endlessly to make them work with Docker, but installing chrome and its drivers it's a pain and I did not get to an end of it. They are still runnable locally, but the data are still needed.